### PR TITLE
[data] un-whitelist a bunch of fixed api documentations

### PR DESCRIPTION
**Pull request by can-anyscale**
- Un-whitelist a bunch of api policy check; these methods are all fixed now
- A previous PR added the `oss` tag to the lint check, caused it to skip in microcheck/premerge because the determine_test_to_run doesn't not produce this `oss` tag. Add a `any_change` tag to go with `oss` tag, for jobs that we want to run everywhere.

Test:
- CI

### CODE REVIEW

File: .buildkite/lint.rayci.yml [hunk 1]
```diff
@@ -23,6 +23,7 @@ steps:
   - label: ":lint-roller: lint: {{matrix}}"
     tags:
       - oss
+      - any_change
```
Comment (bveeramani) [inline .buildkite/lint.rayci.yml:26]: For my own understand, why is this change needed?

**Review by bveeramani — APPROVED** [review]

File: .buildkite/lint.rayci.yml (hunk 1 above)
Comment (can-anyscale) [reply .buildkite/lint.rayci.yml:26]: Previously these lint has no tag associated, which translates to `run these jobs on any change`. One a tag `X` is added, CI will run the job only if PR changes can produce the `X` tag (defined in the determine_test_to_run.py script).

I need to add the `oss` tag, which is a special tag to skip the job upstream. This cause rayci to skip this job as well however, so I need to add to add another `any_change` tag to tell rayci to run this job on any change on OSS.

Comment (aslonnie) [inline .buildkite/lint.rayci.yml:26]: maybe change this tag to `docscheck` or something? and just skip `docscheck` when we need to?

otherwise, this is effectively an `always`, and it feels a bit weird that it still requires `oss` as a tag.

Comment (can-anyscale) [reply .buildkite/lint.rayci.yml:26]: oh there is an always tag, i didn't know about it

**Review by aslonnie — APPROVED** [review]
